Bottom: 66ef3e8228294dd7d365e69fa7a765f262e520dd
Top:    c871ec93220cf8e0913066b6b00972879bc601d2
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-11 15:54:35 +0000

Refresh of speed-up-initial-clipping-pour


---

diff --git a/src/pour.c b/src/pour.c
index cea5f75..240ae0d 100644
--- a/src/pour.c
+++ b/src/pour.c
@@ -671,7 +671,8 @@ ClearPour (DataTypePtr Data, LayerTypePtr Layer, PourType * pour,
       GROUP_LOOP (Data, group);
       {
         r += r_search (layer->line_tree, &region, NULL, line_sub_callback, &info);
-        subtract_poly (info.accumulate, &info.pg);
+        if (info.accumulate != NULL)
+          subtract_poly (info.accumulate, &info.pg);
         info.accumulate = NULL;
         r += r_search (layer->arc_tree,  &region, NULL, arc_sub_callback,  &info);
         r += r_search (layer->text_tree, &region, NULL, text_sub_callback, &info);
@@ -680,7 +681,8 @@ ClearPour (DataTypePtr Data, LayerTypePtr Layer, PourType * pour,
       END_LOOP;
       r += r_search (Data->via_tree, &region, NULL, pin_sub_callback, &info);
       r += r_search (Data->pin_tree, &region, NULL, pin_sub_callback, &info);
-      subtract_poly (info.accumulate, &info.pg);
+      if (info.accumulate != NULL)
+        subtract_poly (info.accumulate, &info.pg);
       info.accumulate = NULL;
     }
